packages/react/src/components/Dialog/Dialog.tsx

```diff
@@ -177,6 +177,21 @@ class Dialog extends AutoControlledComponent<ReactProps<DialogProps>, DialogStat
     },
   })
 
+  /**
```

Comment:
What do you think if to move this logic to `dialogBehavior`?
Since it is about adding accessibility attributes and `getDefaultAriaLabelBy` is based on `props` which are passed to behavior.
Also, to add `id` to header slot, can be done in behavior

```tsx
atrributes: {
  header: {
   id: ...
}
}
```

And use in component like
```tsx
{Header.create(header, {
            defaultProps: {
              as: 'h2',
              styles: styles.header,
          {...accessibility.attributes.header}
            },
          })}
```

Please let me know what do you think.

Reply:
It's how I started it, but I did not know if we should handle the header ID in the behavior, in the case we had to generate it. If we think it's acceptable, I will move all of it back to the dialog behavior, it's cleaner :)